### PR TITLE
Migrate to CNP Draft Store & Archive serenity reports

### PR DESCRIPTION
**Pull request by johnoneill46**
# Description

Combined my changes (DIV-2725) with Tonys' PR #42 
- Archive serenity reports & Migrate to CNP Draft Store for all envs

- Simply updated draftstore baseurl to point to new CNP url rather than pointing to tactical. 
- Aggregate Serenity Reports are being generated but not archived by Jenkins, so they are not accessible. This code tells Jenkins to archive the serenity outputs after the functional test finishes.

No tests are required as mine is an infra change and for Tonys changes, no tests are required as long as Serenity Reports can be seen after the preview build finishes.

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: automation-test/src/test/resources/application-preview.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ccd.caseDataStore.baseUrl=http://ccd-data-store-api-aat.service.core-compute-aat.internal
 
-draft.store.api.baseUrl=https://preproddraftstorelb.moneyclaim.reform.hmcts.net:4301
+draft.store.api.baseUrl=http://draft-store-service-aat.service.core-compute-saat.internal/
```
Comment (mthava) [inline automation-test/src/test/resources/application-preview.properties]: remove these lines and make it a output.tf variable

File: automation-test/src/test/resources/application-aat.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ccd.caseDataStore.baseUrl=http://ccd-data-store-api-aat.service.core-compute-aat.internal
 
-draft.store.api.baseUrl=https://preproddraftstorelb.moneyclaim.reform.hmcts.net:4301
+draft.store.api.baseUrl=http://draft-store-service-aat.service.core-compute-saat.internal/
```
Comment (mthava) [inline automation-test/src/test/resources/application-aat.properties]: remove these lines and make it a output.tf variable

File: automation-test/src/test/resources/application-aat.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ccd.caseDataStore.baseUrl=http://ccd-data-store-api-aat.service.core-compute-aat.internal
 
-draft.store.api.baseUrl=https://preproddraftstorelb.moneyclaim.reform.hmcts.net:4301
+draft.store.api.baseUrl=${draft.store.api.baseUrl}
```
Comment (mthava) [inline automation-test/src/test/resources/application-aat.properties]: this line is not needed

File: automation-test/src/test/resources/application-preview.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ccd.caseDataStore.baseUrl=http://ccd-data-store-api-aat.service.core-compute-aat.internal
 
-draft.store.api.baseUrl=https://preproddraftstorelb.moneyclaim.reform.hmcts.net:4301
+draft.store.api.baseUrl=${draft.store.api.baseUrl}
```
Comment (mthava) [inline automation-test/src/test/resources/application-preview.properties]: this line is not needed

File: infrastructure/output.tf [hunk 5]
```diff
@@ -9,3 +9,7 @@ output "vaultUri" {
 output "idam_s2s_url" {
   value = "http://${var.idam_s2s_url_prefix}-${var.env}.service.${local.ase_name}.internal"
 }
+
+output "draft_store_service_url" {
+    value = "http://draft-store-service-${local.local_env}.service.core-compute-${local.local_env}.internal"
```
[the remark this answers is not in the document]
Comment (johnoneill46) [reply infrastructure/output.tf]: Yep totally correct, good call 👍

**Pull request merged**